microprofile/WFLY-10709-MicroProfile-Tracing-Extension.adoc

```diff
@@ -0,0 +1,93 @@
+= [WFLY-10709] Support Eclipse MicroProfile OpenTracing
+:author:            Juraci Paixão Kröhling
+:email:             jpkroehling@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+:keywords:          observability,microprofile,tracing,openshift
+
+== Overview
+
+In order to improve Observability of applications deployed in WildFly it would be beneficial to provide
+transparent tracing of Java EE components via automatic initialization of a tracer (Jaeger) plus a set
+of instrumentation libraries for components commonly used in Java EE applications, such as JAX-RS and CDI.
+
+This will be implemented by supporting Eclipse MicroProfile OpenTracing 1.1 for applications deployed in WildFly.
+The integration uses the `smallrye-opentracing` component to provide the MicroProfile OpenTracing implementation.
+
+For WildFly 14, this shall be marked as "Tech Preview", to get community feedback before committing to a stable API.
+Notably, this first version lacks any configuration option following WildFly's standard management model, as this is
+something that will likely change before a stable version. 
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-10709[WFLY-10709]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/EAP7-1027[EAP7-1027]
+* https://issues.jboss.org/browse/CLOUD-2728[CLOUD-2728] - Expose EAP "Tracing" information so that it can be consumed by OpenShift Console
+* https://issues.jboss.org/browse/TRACING-85[TRACING-85] - Create jboss-modules for OpenTracing and Jaeger
+* https://issues.jboss.org/browse/TRACING-87[TRACING-87] - OpenTracing integration for JAX-RS in Wildfly
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:tremes@redhat.com[Tomas Remes]
+* mailto:jstourac@redhat.com[Jan Stourac]
+
+=== Affected Projects or Components
+
+The implementation of Eclipse MicroProfile OpenTracing is provided by https://github.com/smallrye/smallrye-opentracing[smallrye-opentracing].
+
+=== Other Interested Projects
+
+* None
+
+== Requirements
+
+* Support https://github.com/eclipse/microprofile-opentracing/blob/1.1/spec/src/main/asciidoc/microprofile-opentracing.asciidoc[Eclipse MicroProfile OpenTracing 1.1] for applications deployed in WildFly, including JAX-RS and CDI integration. Passing the TCK is required.
+* Support a concrete tracer, such as Jaeger's
+* Allow applications to supply their own tracers via `TracerResolver`
+* Each individual application deployed (WAR) should have its own `Tracer` instance
+* Each WAR within an EAR should be treated as an individual WAR, having their own `Tracer` instance
+
+== Nice-to-Have Requirements
+
+Items that are currently being considered for the next version of this feature:
+
+* Enable/disable tracing on the entire server
+* Enable/disable tracing on a per-deployment basis
+* Configuration following WildFly's standard management model
```

Comment:
@jpkrohling, there are some other things that should be considered for full support (and not just from the WildFly side). Even in case they don't make sense from the point of Jaeger server, we need to discuss them and clarify what will happen in such situation:
* What happens when Jaeger server is not available to push messages to (shall we log some warning?), is it possible to provide some kind of probe for administrator to check that connection between client and server is working and messages can be successfully delivered to Jaeger server?
* Is there any kind of HA from the Jaeger server available (this is not a WildFly concern, but we still need a fully featured integration)? If so are we able to detect fail-over and redirect messages to different server?
* Will there be a possibility for the tracing client to use TCP instead of UDP?

Could you please update this analysis with these bullets (feel free to reformulate) or put some statements regarding to those? Thank you!

Reply:
I wasn't sure where this content would be suitable, so, I added to the overview. Could you please check? Let me know if further clarification is needed.

Reply:
Well, my first thought was to add a bullets here in the "Nice to have" section (nice to though about), but generally I have no strong opinion. Anyway, I've just checked and I am not sure whether I found updated content in the overview section?

Reply:
I'm sorry, looks like I didn't push the new commit... I just opened a new PR to add this: https://github.com/wildfly/wildfly-proposals/pull/123

Reply:
Thank you, I'll check there.